src/bin/pg_autoctl/cli_service.c

```diff
@@ -460,29 +465,8 @@ cli_service_status(int argc, char **argv)
 
 	if (outputJSON)
 	{
-		JSON_Value *js = json_value_init_object();
-		JSON_Value *jsPostgres = json_value_init_object();
-
-		JSON_Value *jsPGAutoCtl = json_value_init_object();
-		JSON_Object *jsobj = json_value_get_object(jsPGAutoCtl);
-
-		JSON_Object *root = json_value_get_object(js);
+		bool includeStatus = true;
 
-		/* prepare both JSON objects */
-		json_object_set_number(jsobj, "pid", (double) pid);
-
-		if (!pg_setup_as_json(pgSetup, jsPostgres))
-		{
-			/* can't happen */
-			exit(EXIT_CODE_INTERNAL_ERROR);
-		}
-
-		/* concatenate JSON objects into a container object */
-		json_object_set_value(root, "postgres", jsPostgres);
-		json_object_set_value(root, "pg_autoctl", jsPGAutoCtl);
-
-		(void) cli_pprint_json(js);
+		(void) fprint_pidfile_as_json(pathnames->pid, includeStatus);
```

Comment:
The new json format is missing all the information about the actual postgres process, all you see is info about the postgres service.
new:
```json
{
  "pid": 5599,
  "status": "running",
  "pgdata": "monitor",
  "version": "1.3",
  "semId": 655360,
  "services": [
    {
      "name": "postgres",
      "pid": 5603,
      "status": "running",
      "version": "1.3",
      "pgautofailover": "1.3"
    },
    {
      "name": "listener",
      "pid": 5604,
      "status": "running",
      "version": "1.3",
      "pgautofailover": "1.3"
    }
  ]
}
```

old
```json
{
  "postgres": {
    "pgdata": "/home/jelte/work/pg_auto_failover/monitor",
    "pg_ctl": "/home/jelte/.pgenv/pgsql/bin/pg_ctl",
    "version": "12.3",
    "host": "/tmp",
    "port": 5500,
    "proxyport": 0,
    "pid": 5608,
    "in_recovery": false,
    "control": {
      "version": 0,
      "catalog_version": 0,
      "system_identifier": "0"
    },
    "postmaster": {
      "status": "ready"
    }
  },
  "pg_autoctl": {
    "pid": 5599
  }
}
```

An easy way to fix this would be to make the JSON instead be like this:
```js
{
  "postgres": {
     // all postgres process values from the old one
  },
  "pg_autoctl": {
    // the pidfile data
  }
}
```

This would actually make the new JSON format compatible with the old format, since it only adds fields and doesn't remove them.

Reply:
Thanks, good thinking! Done now.